language/tools/move-package/src/lib.rs

```diff
@@ -168,21 +166,17 @@ impl BuildConfig {
         writer: &mut W,
     ) -> Result<CompiledPackage> {
         let resolved_graph = self.resolution_graph_for_package(path, writer)?;
-        let mutx = PackageLock::lock();
-        let ret = BuildPlan::create(resolved_graph)?.compile_no_exit(writer);
-        mutx.unlock();
-        ret
+        let _mutx = PackageLock::lock();
+        BuildPlan::create(resolved_graph)?.compile_no_exit(writer)
     }
 
     #[cfg(feature = "evm-backend")]
     pub fn compile_package_evm<W: Write>(self, path: &Path, writer: &mut W) -> Result<()> {
         // resolution graph diagnostics are only needed for CLI commands so ignore them by passing a
         // vector as the writer
         let resolved_graph = self.resolution_graph_for_package(path, &mut Vec::new())?;
-        let mutx = PackageLock::lock();
-        let ret = BuildPlan::create(resolved_graph)?.compile_evm(writer);
-        mutx.unlock();
-        ret
+        let _mutx = PackageLock::lock();
+        BuildPlan::create(resolved_graph)?.compile_evm(writer)
```

Comment:
Would it be safer to do something like this?
```suggestion
        let mutx = PackageLock::lock();
        BuildPlan::create(resolved_graph)?.compile_evm(writer)
        drop(mutx)
```

Reply:
Talked about this offline, and settled on commenting the `lock()` call instead -- the issue with the explicit `drop` is similar to the explicit `unlock` in that it gives the false impression that all exit paths flow through that point in the function, when that isn't true (e.g. the return edge for the `?` will not flow through the explicit `drop`).